### PR TITLE
Eeitt authentication

### PR DESCRIPTION
**Pull request by DanielConnelly**

### CODE REVIEW

File: app/uk/gov/hmrc/eeittadminfrontend/config/ApplicationLoader.scala [hunk 1]
```diff
@@ -247,13 +249,12 @@ trait ApplicationModule extends BuiltInComponents
 
   val eeittUrl: String = s"${baseUrl("eeittadmin")}/eeitt"
 
-
-
+  val clientID = configuration.getString("ClientID")
```
Comment (pawelpanasewicz) [inline app/uk/gov/hmrc/eeittadminfrontend/config/ApplicationLoader.scala]: There is bunch of nice libraries which allows to create configs in type safe manner. 
We could consider to adopt for example this one: https://github.com/pureconfig/pureconfig

File: app/uk/gov/hmrc/eeittadminfrontend/services/GoogleVerifier.scala [hunk 2]
```diff
@@ -21,14 +21,25 @@ import java.util.Collections
...
 
-  def apply(string: String) = {
-    tokenVerifier.verify(string).getPayload.getEmail
+  def apply(token: String): String = {
+    tokenVerifier.verify(token).getPayload.getEmail
   }
 }
+
+
+object GoogleVerifier {
```
Comment (pawelpanasewicz) [inline app/uk/gov/hmrc/eeittadminfrontend/services/GoogleVerifier.scala]: Looks like I've found dead code ;)

File: app/uk/gov/hmrc/eeittadminfrontend/services/AuthService.scala [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+package uk.gov.hmrc.eeittadminfrontend.services
+
+import cats.data.Validated
+import play.api.Configuration
+import uk.gov.hmrc.eeittadminfrontend.models.{Email, LoginError}
+
+class AuthService(config: Configuration) {
+
+  def checkUser(email: Email): Validated[LoginError, Unit] = {
```
Comment (pawelpanasewicz) [inline app/uk/gov/hmrc/eeittadminfrontend/services/AuthService.scala]: Do we really want to mix `Validated` and `Eithers`? In most places we tend to use either. If we start using one and another we might end up with convertions from one format to another. I would suggest to stick to only one data type representing possible failures. To be discussed:
 
Howver if we would like to use Validated I suggest to use nicer syntax so this code could look like:
```scala
import cats.data._
import cats.syntax.all._ //or cats.syntax.validted._
if(users.contains(email.value)) LoginError("Unauthorised User").invalid else ().valid
```

**Comment by pawelpanasewicz** [comment]
nice stat :+1:  +76 −161

File: app/uk/gov/hmrc/eeittadminfrontend/services/AuthService.scala (hunk 3 above)
Comment (DanielConnelly) [inline app/uk/gov/hmrc/eeittadminfrontend/services/AuthService.scala]: I used either as neither of the results is invalid. they are both valid responses. didnt feel right labelling one as invalid when it is ok.

Comment (DanielConnelly) [inline app/uk/gov/hmrc/eeittadminfrontend/services/AuthService.scala]: Ah I just got what you meant. I am kinda just using validated to play about with cats a little.

Comment (pawelpanasewicz) [inline app/uk/gov/hmrc/eeittadminfrontend/services/AuthService.scala]: Have fun :) We can discuss later what are trade-offs using validated/either/or custom validated which happens for example [in gform](https://github.com/hmrc/gform/blob/349a3a1c0156ce24845f173a054e780d22baaed7/app/uk/gov/hmrc/gform/core/ValidationResult.scala).

As a quick hint - Validated are good for collecting error messages but you need to use applicative builder in order to have it.

File: app/uk/gov/hmrc/eeittadminfrontend/services/AuthService.scala [hunk 4]
```diff
@@ -16,18 +16,16 @@
...
+  val validUserList: Array[String] = config.getString("basicAuth.user").getOrElse("").split(":")
+
   def checkUser(email: Email): Validated[LoginError, Unit] = {
-    val validUserList = config.getString("basicAuth.user").getOrElse("").split(":")
-    if(validUserList.contains(email.value)) {
-      Validated.invalid(LoginError("Unauthorised User"))
-    } else {
-      Validated.valid(())
-    }
+    if(validUserList.contains(email.value)) LoginError("Unauthorised User").invalid else ().valid
```
Comment (pawelpanasewicz) [inline app/uk/gov/hmrc/eeittadminfrontend/services/AuthService.scala]: beautiful, isn't it :+1:

**Comment by pawelpanasewicz** [comment]
LGTM

**Pull request merged**